### ISSUE TITLE
html tags in research focus field

### ISSUE THREAD

**Issue by jfurfey** [BibApp/BibApp#80]
The text field for research focus used to accept some basic html tags such as `<p>, <ul>, <ol>`, etc.

It no longer does, they are rendered as text.

**Comment by hading** [comment]
I'll take a look. I would have thought that this would have showed up when the conversion to Rails 3 happened, as it sounds like an html safety issue and I'd expect what happens now is that it just gets quoted by default. So I'm not sure. 

To really fix it I'll need to do a bit of research. I'm pretty sure there are gems out there to allow one to whitelist certain tags and quote other ones and that's really the safest way to go with something like this. But I don't know what's available right now.

**Comment by hading** [comment]
I've fixed part of this issue (and will close it). I installed the loofah and loofah-activerecord gems. This field in the Person model is now marked as an html fragment which will (before validation/saving) be sanitized, quoting html elements that are considered dangerous (when you go back and edit again you'll actually see the quoted form). I've also marked the field in the people#show view (actually this goes down to the show_research_focus partial) as html_safe.

If this is shown elsewhere and still shows up incorrectly let me know and/or start another ticket and I'll fix that place. If there are other fields where people would like to be able to use some HTML let me know and I'll do the same thing with those. If there are other fields that are already being shown as HTML safe but not passing though this method we should use something like this to quote them as appropriate. So if anyone comes upon that let me know, but I may look for (and start an issue for) them anyway.
